docs/user-guide/api-mediation/api-mediation-modulith.md

```diff
@@ -0,0 +1,143 @@
+* Minimize JVM processes.
+* Minimize networking operations.
+* Unified configuration options.
+
+## Architecture
+
+This section contains an overview of the architecture of the API Mediation Layer single-service deployment mode.
+
+![Zowe API ML Single-service Architecture Diagram](../../images/common/zowe-architecture-apiml-single-service.png)
+<!-- TODO
+Diagram
+Data flows
+Use the example from the current architecture diagram?
+ -->
+
+## Breaking Changes
+
+:::note
+The following instructions assume a default address space prefix `ZWE1`. Update as needed.
+:::
+
+To run API Mediation Layer in single-service mode, the system programmer is required to make configuration changes in the following areas:
+
+* Port usage is shared in a single address space. <!-- I think we should remove "is shared in a single address spece" as this information should be detailed in the corresponding section below.-->
+* Log prefix is unified for all components. <!--Similarly, "is unified for all components" should be explained in the corresponding section. -->
+* AT-TLS rule adaptations.
+
+### Port usage
+
+The new single-service deployment mode runs all API Mediation Layer components in a single JVM process. <!-- Please add a single sentence about the benefit of using a single JVM process.-->For backward compatibility reasons, this single process handles connections to both the Gateway Service and the Discovery Service ports (defaults 7554 and 7553).
+
+The single-service API Mediation Layer address space uses ports defined in `components.gateway.port` and `components.discovery.port`.
+
+Update the network permissions to reflect this change. Both ports will be under z/OS address space `ZWE1AG`. <!-- Can we please include a codeblock example of this configuration? -->
+
+### Log Prefix
+
+Logs from internal API Mediation Layer components such as the Discovery Service, API Catalog, Caching Service will appear under the `ZWE1AG` prefix.
+<!-- Can we please include a log example? -->
+
+### AT-TLS
+
+If the installation is configured with AT-TLS, rules need to be updated. Perform the following updates to the PAGENT rules:
+
+* Update job name filters to use `ZWE1AG`.
+* Remove unneeded rules that were performing the handling.
+
+<!--We need to include an example of these PAGENT rules configuration -->
+
+**Note:** TCP HTTP calls are still in use for high availability scenarios to maintain synchronization between instances accross LPARs.
+
+## Limitations
+
+The following features are not supported in the technical preview release of the single service API Mediation Layer:
+
+* Multi tenancy deployment is not supported.
+* Docker container deployments.
+
+## Enable the Single-service API Mediation Layer
+
+To switch the API Mediation Layer into modularized mode, perform the following changes to the installation's `zowe.yaml` file:
+
+1. Add component `apiml` and enable it: <!-- Please add where specifically in the yaml. -->
+
```

Comment:
@janan07 Please add the inline comments as github comments.
About the location, what do you mean? `components` is a top level property, so it doesn't really matter where in the file they set it, as long as users follow YAML conventions. This I think it's outside the scope of this documentation